questionpy/_qtype.py

```diff
@@ -36,9 +35,11 @@ class Question(ABC):
     options: FormModel
     question_state: BaseQuestionState
 
-    options_class: ClassVar[type[FormModel]]
-    question_state_class: ClassVar[type[BaseQuestionState]]
-    question_state_with_version_class: ClassVar[type[QuestionStateWithVersion]]
+    options_class: ClassVar[type[FormModel]] = reify_type_hint("options", FormModel)
+    question_state_class: ClassVar[type[BaseQuestionState]] = reify_type_hint("question_state", BaseQuestionState)
+    question_state_with_version_class: ClassVar[type[QuestionStateWithVersion]] = cached_class_property(
+        lambda cls: QuestionStateWithVersion[cls.options_class, cls.question_state_class]  # type: ignore[name-defined]
+    )
```

Comment:
NB: Keine Verwendung als Decorator (`@cached_class_property`), weil Mypy das nur versteht, wenn auch `@classmethod` dran ist, was dann wieder support in `cached_class_property` braucht, etc. So war es einfacher.